controller/search.js

```diff
@@ -16,13 +15,9 @@ function setup( backend, query ){
       body: query( req.clean )
     };
 
-    if (req.clean.layers) {
-      cmd.type = req.clean.layers;
-    }
-
-    // set type if input suggests targeting a layer(s)
-    if (req.clean.default_layers_set && req.clean.parsed_input) {
-      cmd.type = req.clean.parsed_input.target_layer || cmd.type;
+    if (req.clean.type !== undefined) {
+      cmd.type = req.clean.type;
+      delete req.clean.type; // remove type from clean to avoid clutter
```

Comment:
We currently only show `req.clean` object as part of the result _geocode-json_, so if we want to show the client which layers we decided to query, it might need to be set in both places. We should still be returning `layers` array so this might not be super important.